python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -100,14 +100,17 @@ def _apply_nms(orig_prediction: dict, iou_threshold: float = 0.5):
     :return: Model prediction after nms is applied
     :rtype: dict
     """
-    keep = torchvision.ops.nms(orig_prediction[BOXES],
-                               orig_prediction[SCORES],
-                               iou_threshold)
-
     nms_prediction = orig_prediction
-    nms_prediction[BOXES] = nms_prediction[BOXES][keep]
-    nms_prediction[SCORES] = nms_prediction[SCORES][keep]
-    nms_prediction[LABELS] = nms_prediction[LABELS][keep]
+    boxes = orig_prediction[BOXES]
+    scores = orig_prediction[SCORES]
+    if boxes.shape[0] != 0:
```

Comment:
```suggestion
    if boxes.numel() == 0:
```
Suspecting this pytorch operation may be more reliable..?

Reply:
I think you meant:
```
boxes.numel() != 0:
```
but sure, updated to this, it's the same functionally